### ISSUE TITLE
Incorrect metrics data from sinks datadog_metric

### ISSUE THREAD

**Issue by bachrilq** [vectordotdev/vector#13846]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

So I compare the resulting metric between `Datadog Agent` and `Datadog Agent through Vector`. The details are like this:

- Metrics > Datadog Agent > Datadog Server
- Metrics > Datadog Agent > Vector > Datadog Server

and I found Incorrect data between the two methods. If the metrics send through `Vector`, the metrics will multiply about 10X. As shown in the picture below, If only using the Datadog agent the metrics are only 50 but If using `Vector` the metric will increase to 500.

- Without Vector
<img width="577" alt="Screen Shot 2022-08-04 at 15 48 05" src="https://user-images.githubusercontent.com/40727644/182805701-87875f90-e522-46d7-8d71-e53db6f153b4.png">

- Using Vector
<img width="621" alt="Screen Shot 2022-08-04 at 15 48 10" src="https://user-images.githubusercontent.com/40727644/182805691-a21038c2-1e15-491b-aa4c-f484578147a7.png">

### Configuration

```text
[sources.datadog_agent_forwarder]
type = "datadog_agent"
address = "127.0.0.1:8128"
disable_logs = true
disable_traces = true
multiple_outputs = true


[sinks.datadog_metric]
type = "datadog_metrics"
inputs = ["datadog_agent_forwarder.metrics"]
api_key = "XXXXXXXXXXXXX"
request.concurrency = "adaptive"
```

### Version

vector 0.22.4

**Comment by tobz** [comment]
👋🏻 

This is almost certainly related to https://github.com/vectordotdev/vector/issues/9969, which we just recently pushed a fix for. Would you be able/willing to test out a nightly build to see if that resolves the issue for you?

Unfortunately this fix isn't yet part of a standard release.

**Comment by bachrilq** [comment]
Hi @tobz , thank you for your reply, Yes of course absolutely, where I can get the nightly build release?

**Comment by tobz** [comment]
@bachrilq You can head to https://vector.dev/download/ and select "nightly" from the drop-down.

**Comment by bachrilq** [comment]
Noted, thank you for the information, I'll let you know the result after testing the nightly release

**Comment by bachrilq** [comment]
Hi @tobz , After I tried the nightly release, thank you for the issue is fixed. But unfortunately, I got another Issue regarding `filter` in `transforms`, so I tried the config in a lower version like in `0.22.3` and below, the `filter` is working properly, but I don't know why since in version `0.23` and nightly release, the filter is not working properly. For that issue, is better if I create another Open Issue, or is ok if I explain the issue here? , Thank You

**Comment by tobz** [comment]
I see you opened a separate issue, which is exactly what I was going to suggest to do. 👍🏻

**Comment by jszwedko** [comment]
Closed by https://github.com/vectordotdev/vector/pull/13394

**Comment by bachrilq** [comment]
Hi @jszwedko, Is these changes shipped with the latest stable release at version v0.23.3? or still on a nightly build release? , Thank you 🙏

**Comment by jszwedko** [comment]
Hi @bachrilq ! This is unreleased but will be included in v0.24.0 which we are targeting for next week.

**Comment by bachrilq** [comment]
Alright, Glad to hear that! , Thank you for your information @jszwedko 🙏
